### PR TITLE
PVP Leaderboard #3

### PR DESCRIPTION
**Pull request by Houstonruss**
Dev to master merger.

### CODE REVIEW

**Review by botn365 — APPROVED** [review]
lgtm

File: src/main/java/gregtech/api/metatileentity/BaseMetaPipeEntity.java
Comment (basdxz) [inline src/main/java/gregtech/api/metatileentity/BaseMetaPipeEntity.java:1]: Evaluate trycaches? Should keep them?

File: src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java
Comment (basdxz) [inline src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java:1]: Ditto.

File: src/main/java/gregtech/api/util/GT_ModHandler.java [hunk 1]
```diff
@@ -1623,7 +1645,8 @@ public static List<ItemStack> getRecipeOutputsBuffered(ItemStack... aRecipe) {
      * Gives you a list of the Outputs from a Crafting Recipe
      * If you have multiple Mods, which add Bronze Armor for example
      */
-    public static List<ItemStack> getRecipeOutputs(List<IRecipe> aList, boolean aDeleteFromList, ItemStack... aRecipe) {
+    @SneakyThrows
```
Comment (basdxz) [inline src/main/java/gregtech/api/util/GT_ModHandler.java:1648]: Do we want sneaky throws here?

File: src/main/java/gregtech/common/GT_Worldgenerator.java [hunk 2]
```diff
@@ -507,6 +509,14 @@ public void run() {
             if (tChunk != null) {
                 tChunk.isModified = true;
 
+                if (debugWorldGen && usedOreMix == null) {
+                    for (val player : this.mWorld.playerEntities) {
+                        val message = String.format("No orevein selected for chunk: %s", tChunk.getChunkCoordIntPair());
```
Comment (basdxz) [inline src/main/java/gregtech/common/GT_Worldgenerator.java:514]: Ensure this does not spam log in release!

File: src/main/java/gregtech/common/misc/explosions/GT_Explosion.java [hunk 3]
```diff
@@ -179,7 +179,8 @@ protected void doEntityStuff() {
                     disX /= disMag;
                     disY /= disMag;
                     disZ /= disMag;
-                    blockDensity = pubWorld.getBlockDensity(expVec, entity.boundingBox);
+//                    blockDensity = pubWorld.getBlockDensity(expVec, entity.boundingBox);
+                    blockDensity = 1.0 / 2.0;
```
Comment (basdxz) [inline src/main/java/gregtech/common/misc/explosions/GT_Explosion.java:183]: Wait, does this ignore block hardness??

File: src/main/java/gregtech/loaders/preload/metatileentity/FluidPipeLoader.java
Comment (basdxz) [inline src/main/java/gregtech/loaders/preload/metatileentity/FluidPipeLoader.java:1]: Who did this?

File: src/main/java/gregtech/common/GT_Worldgenerator.java (hunk 2 above)
Comment (Houstonruss) [reply src/main/java/gregtech/common/GT_Worldgenerator.java:514]: I FIXED THIS
AND I CONFIRMED IT WORK
[divine code]

**Pull request merged**